adoc/chapters/programming_interface.adoc

```diff
@@ -17945,43 +17945,77 @@ device_event(___unspecified___)
 [[sec:atomic-references]]
 === Atomic references
 
-The SYCL specification provides atomic operations based on the library
-syntax from the next {cpp} specification. The set of supported orderings is
+The [code]#sycl::atomic_ref# class provides the ability to perform atomic
+operations in device code with a syntax similar to the {cpp} standard
+[code]#std::atomic_ref#.  The [code]#sycl::atomic_ref# class must not be used
+in host code.
+
+Unlike [code]#std::atomic_ref#, [code]#sycl::atomic_ref# does not provide a
+default memory ordering for its operations.  Instead, the application must
+specify a default ordering via the [code]#DefaultOrder# template parameter.
+This ordering is used as a default for most of the atomic operations, but
+each member function also provides an optional parameter that allows the
```

Comment:
Do you think we need to state explicitly that the arithmetic operators always use the default and are always read-modify-write operations?  Or is that clear enough from the way we define them in the table (e.g. that `operator++` is equivalent to `fetch_add(1)`)?

Reply:
I think we can rely on the description in the table for this level of detail.

I could change the wording to "most member functions also provide ..." if we think that's better.  I guess it depends on whether the term "member function" also includes the operators.

Reply:
Yeah, that's what I was unsure of.  I did some digging and it looks like an "operator function" declared inside a class is considered a "member function" (https://eel.is/c++draft/over.oper.general#7).

Your suggested fix sounds good.